File: src/main/java/com/web/rpg/service/player/impl/PlayerServiceImpl.java
package com.web.rpg.service.player.impl;

import com.web.rpg.model.Characters.PlayerCharacter;
import com.web.rpg.service.character.CharacterService;
import com.web.rpg.service.player.PlayerService;
import lombok.AllArgsConstructor;

import java.util.UUID;

@AllArgsConstructor
public class PlayerServiceImpl implements PlayerService {

    private final CharacterService characterService;

    @Override
    public Integer getCountOfManaBottles(UUID id) {
        if (id == null) {
            return null;
        }
        PlayerCharacter character = characterService.findById(id);
        return character.getCountOfBigManaPointBottles() +
                character.getCountOfMiddleManaPointBottles() +
                character.getCountOfSmallManaPointBottles();
    }

    @Override
    public Integer getCountOfHitPointBottles(UUID id) {
        if (id == null) {
            return null;
        }
        PlayerCharacter character = characterService.findById(id);
        return character.getCountOfBigHitPointBottle() +
                character.getCountOfMiddleHitPointBottle() +
                character.getCountOfSmallHitPointBottle();
    }
}

File: src/main/java/com/web/rpg/service/player/impl/PlayerServiceImpl.scala
package com.web.rpg.service.player.impl

import java.util.UUID

import com.web.rpg.service.character.CharacterService
import com.web.rpg.service.player.PlayerService
import org.springframework.beans.factory.annotation.Autowired

class PlayerServiceImpl @Autowired()(val characterService: CharacterService) extends PlayerService {

  override def getCountOfManaBottles(id: UUID): Integer = {
    if (id == null) return null
    val character = characterService.findById(id)
    character.getCountOfBigManaPointBottles +
      character.getCountOfMiddleManaPointBottles +
      character.getCountOfSmallManaPointBottles
  }

  override def getCountOfHitPointBottles(id: UUID): Integer = {
    if (id == null) return null
    val character = characterService.findById(id)
    character.getCountOfBigHitPointBottle +
      character.getCountOfMiddleHitPointBottle +
      character.getCountOfSmallHitPointBottle
  }

}
